src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -409,6 +409,29 @@ def reverse_resolve(self, use_case_id: UseCaseID, org_id: int, id: int) -> Optio
         string: str = obj.string
         return string
 
+    def bulk_reverse_resolve(
+        self, use_case_id: UseCaseID, org_id: int, ids: Collection[int]
```

Comment:
`Collection` allows one to pass a `Set` but I think you pretty much want to enforce that there's ordering in the ids you pass. Alternatively you can also return a `Mapping[int, Optional[str]]` to avoid this pitfall

Reply:
Very good point, I was thinking that all I need is to get the size and an iterator  but of course that doesn't guarantee the order.  
Thanks for the input !